cg/cli/workflow/balsamic/base.py

```diff
@@ -1,323 +1,263 @@
-            original_fastq_path = Path(fastq_data["path"])
-            linked_fastq_name = context.obj["fastq_handler"].FastqFileNameCreator.create(
-                lane=fastq_data["lane"],
-                flowcell=fastq_data["flowcell"],
-                sample=link_obj.sample.internal_id,
-                read=fastq_data["read"],
-                more={"undetermined": fastq_data["undetermined"]},
-            )
-            concatenated_fastq_name = context.obj[
-                "fastq_handler"
-            ].FastqFileNameCreator.get_concatenated_name(linked_fastq_name)
-            linked_fastq_path = wrk_dir / linked_fastq_name
-            linked_reads_paths[fastq_data["read"]].append(linked_fastq_path)
-            concatenated_paths[fastq_data["read"]] = f"{wrk_dir}/{concatenated_fastq_name}"
-
-            if linked_fastq_path.exists():
-                LOG.info("found: %s -> %s", original_fastq_path, linked_fastq_path)
-            else:
-                LOG.debug("destination path already exists: %s", linked_fastq_path)
-
-        if link_obj.sample.is_tumour:
-            tumor_paths.add(concatenated_paths[1])
-        else:
-            normal_paths.add(concatenated_paths[1])
-
-        if not target_bed:
-            target_bed_filename = get_target_bed_from_lims(
-                context.obj["lims_api"], context.obj["store_api"], link_obj.sample.internal_id
-            )
-            target_beds.add(target_bed_filename)
-
-    if len(application_types) != 1:
-        raise BalsamicStartError(
-            "More than one application found for this case: %s" % ", ".join(application_types)
-        )
-
-    if not application_types.issubset(acceptable_applications):
-        raise BalsamicStartError("Improper application for this case: %s" % application_types)
-
-    nr_paths = len(tumor_paths) if tumor_paths else 0
-    if nr_paths != 1:
-        raise BalsamicStartError("Must have exactly one tumor sample! Found %s samples." % nr_paths)
-
-    tumor_path = tumor_paths.pop()
-
-    normal_path = None
-    nr_normal_paths = len(normal_paths) if normal_paths else 0
-
-    if nr_normal_paths == 1:
-        normal_path = normal_paths.pop()
-    elif nr_normal_paths > 1:
-        raise BalsamicStartError("Too many normal samples found: %s" % nr_normal_paths)
-
-    if target_bed and not application_types.issubset(applications_requiring_bed):
-        raise BalsamicStartError(
-            "--target_bed is incompatible with %s" % " ".join(application_types)
+def config_case(context, panel_bed, case_id, dry):
+    """Create config file for BALSAMIC analysis for a given CASE_ID"""
+
+    LOG.info(f"Creating config file for {case_id}.")
+    case_object = context.obj["BalsamicAnalysisAPI"].get_case_object(case_id)
+    if not case_object:
+        LOG.warning(f"{case_id} not found!")
+        raise click.Abort()
+    if not case_object.links:
+        LOG.warning(f"{case_id} number of samples is {len(case_object.links)}!")
```

Comment:
This one has been a struggle since I too would prefer not to use 'if not' statements. Alternative for this would be (in some cases in balsamic cli) 4 level nested if statement, which I tried and felt was even less readable. Could you point me at some good examples in CG where this is done properly?